rust/src/delta.rs

```diff
@@ -1034,6 +1058,34 @@ impl DeltaTable {
         })
     }
 
+    /// Create a DeltaTable with version 0 given the provided MetaData and Protocol
+    pub async fn create(
+        &mut self, 
+        metadata: DeltaTableMetaData, 
+        protocol: action::Protocol
+    ) -> Result<(), DeltaTableError> {
+        
+        let meta = action::MetaData::try_from(metadata)?;
+
+        // TODO add commit info action
+        let actions = vec![
+            Action::protocol(protocol), 
+            Action::metaData(meta)
+        ];
+        
+        let mut transaction = self.create_transaction(None);
+        transaction.add_actions(actions);
+
+        // Need better error handling here
+        let prepared_commit = transaction.prepare_commit(None).await.unwrap();
+        self.try_commit_transaction(&prepared_commit, 0).await.unwrap();
```

Comment:
These two lines are still using `unwrap()`, they return `DeltaTransactionError` on Err, but it has `From` implemented on `DeltaTableError`. When I tried to add a `From` for `DeltaTableError` to `DeltaTransactionError` I got some compile issue because of a cycle. Wondering how I can handle possible errors here better?

Reply:
This is a good question. I feel like the the error wrapping might have been done in the wrong direction. I think it makes more sense to make `DeltaTransactionError` an enum variant of `DeltaTableError`, not the other way around. @xianwill @rtyler @mosyp what do you think?

Reply:
Maybe if I return a `DeltaTransactionError` here instead of a `DeltaTableError`?

Reply:
Yeah, returning `DeltaTransactionError` here will work, but it just feels a little bit awkward because some of the errors are not related to transactions. Let's do this for now, we can manage the `DeltaTableError` change later as a follow up task.

Reply:
created a follow up task https://github.com/delta-io/delta-rs/issues/346